interfaces/IBF-dashboard/src/app/services/map.service.spec.ts

```diff
@@ -1,28 +1,102 @@
-import { provideHttpClientTesting } from '@angular/common/http/testing';
-import { TestBed } from '@angular/core/testing';
-import { RouterTestingModule } from '@angular/router/testing';
-import { TranslateModule } from '@ngx-translate/core';
-import { MapService } from './map.service';
 import {
   provideHttpClient,
   withInterceptorsFromDi,
 } from '@angular/common/http';
+import { provideHttpClientTesting } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+import { RouterModule } from '@angular/router';
+import { TranslateModule } from '@ngx-translate/core';
+import { PlaceCode } from 'src/app/models/place-code.model';
+import { AdminLevelService } from 'src/app/services/admin-level.service';
+import { MapService } from 'src/app/services/map.service';
+import {
+  MOCK_COUNTRYDISASTERSETTINGS,
+  MOCK_LAYERS,
+  MOCK_TRIGGEREDAREAS as MOCK_TRIGGEREDAREAS,
+} from 'src/app/services/map.service.spec.helper';
+import { IbfLayerName } from 'src/app/types/ibf-layer';
 
 describe('MapService', () => {
   let service: MapService;
+  let adminLevelService: AdminLevelService;
 
   beforeEach(() => {
     TestBed.configureTestingModule({
-      imports: [RouterTestingModule, TranslateModule.forRoot()],
+      imports: [RouterModule.forRoot([]), TranslateModule.forRoot()],
       providers: [
         provideHttpClient(withInterceptorsFromDi()),
         provideHttpClientTesting(),
       ],
     });
     service = TestBed.inject(MapService);
+    adminLevelService = TestBed.inject(AdminLevelService);
   });
 
   it('should be created', () => {
     expect(service).toBeTruthy();
   });
+
+  describe('getLayerByName', () => {
+    beforeEach(() => {
+      service.layers = MOCK_LAYERS;
+    });
+
+    it('should get a layer from its name', () => {
+      const layerName = IbfLayerName.waterpointsInternal;
+
+      const expected = MOCK_LAYERS[0];
+
+      expect(service.getLayerByName(layerName)).toEqual(expected);
+    });
+
+    it('should be undefined if no layer has the provided name', () => {
+      const layerName = IbfLayerName.gauges;
+
+      expect(service.getLayerByName(layerName)).toBeUndefined();
+    });
+  });
+
+  describe('getPlaceCodeParent', () => {
+    it('should return the placeCodeParent of the provided placeCode', () => {
+      // eslint-disable-next-line @typescript-eslint/dot-notation
```

Comment:
is it easy to process the rules correctly here? and if not, is it then best practice to use this individual ignores, instead of just setting the rule to 'off' for now?